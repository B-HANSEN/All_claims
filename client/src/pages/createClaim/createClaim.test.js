import { render, screen } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import CreateClaim from '.';

const shallowRender = Component =>
	new ShallowRenderer(Component).getRenderOutput();

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedUsedNavigate,
}));

beforeEach(() => {
	render(<CreateClaim />);
});

describe('Create Claim Page tests', () => {
	it('should match snapshot', () => {
		expect(shallowRender(<CreateClaim />)).toMatchSnapshot();
	});

	it('should contain form labels', () => {
		const policy = screen.getByLabelText(/policy number:/i);
		const holder = screen.getByLabelText(/holder name:/i);
		const item = screen.getByLabelText(/insured item/i);
		const amount = screen.getByLabelText(/claim amount/i);
		const description = screen.getByLabelText(/description/i);
		const date = screen.getByLabelText(/incident date/i);
		const processingFee = screen.getByLabelText(/processing fee/i);

		expect(policy).toBeInTheDocument();
		expect(holder).toBeInTheDocument();
		expect(item).toBeInTheDocument();
		expect(amount).toBeInTheDocument();
		expect(description).toBeInTheDocument();
		expect(date).toBeInTheDocument();
		expect(processingFee).toBeInTheDocument();
	});

	it('should contain a header', () => {
		const formHeading = screen.getByText(/create claim/i);
		expect(formHeading).toBeInTheDocument();
	});

	it('should contain a link back to the claims lists page', () => {
		const navigateToClaimList = screen.getByRole('button', {
			name: /show claim list/i,
		});

		expect(navigateToClaimList).toBeEnabled();
	});

	it('should contain a button to a new claim', () => {
		const navigateToClaimList = screen.getByRole('button', {
			name: /add new claim/i,
		});

		expect(navigateToClaimList).toBeEnabled();
	});
});
