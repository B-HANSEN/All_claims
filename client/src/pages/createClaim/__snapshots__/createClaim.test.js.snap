// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Claim Page tests should match snapshot 1`] = `
<div
  className="CreateClaim"
>
  <Header />
  <section
    className="CreateClaim__MainSection"
  >
    <h1>
      Create Claim
    </h1>
    <ClaimsForm />
  </section>
</div>
`;
